action/protocol/staking/handlers.go

```diff
@@ -581,7 +581,8 @@ func (p *Protocol) handleCandidateUpdate(ctx context.Context, act *action.Candid
 	}
 
 	if err := csm.Upsert(c); err != nil {
-		return nil, errors.Wrapf(err, "failed to put state of candidate %s", c.Owner.String())
+		log.L().Debug("failed to put state of candidate", zap.Error(err))
+		return p.settleAction(ctx, csm, uint64(iotextypes.ReceiptStatus_ErrCandidateConflict), gasFee)
```

Comment:
https://github.com/iotexproject/iotex-core/issues/2141
More cases in this file need to be addressed.